lib/react_client.dart

```diff
@@ -473,9 +490,8 @@ final ReactDartInteropStatics _dartInteropStatics = (() {
 })();
 
 abstract class _ReactDartInteropStatics2 {
-  // TODO 3.1.0-wip expose for testing?
   /// The zone in which all component lifecycle methods are run.
-  static final componentZone = Zone.root;
+  static final componentZone = currentComponentZone;
```

Comment:
This variable only gets initialized once since it's `static`, so it won't respect updates to `currentComponentZone`. We should support that since
- we're not guaranteed that all tests run in the same zone (they could be descendant zones, etc)
- we may want to set it to two different zones during the execution of the program